server/item/splash_potion.go

```diff
@@ -0,0 +1,61 @@
+package item
+
+import (
+	"github.com/df-mc/dragonfly/server/item/potion"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+)
+
+// SplashPotion is an item that grants effects when thrown.
+type SplashPotion struct {
+	// Type is the type of splash potion.
+	Type potion.Potion
+}
+
+// MaxCount ...
+func (s SplashPotion) MaxCount() int {
+	return 1
+}
+
+// Use ...
+func (s SplashPotion) Use(w *world.World, user User, ctx *UseContext) bool {
+	splash, ok := world.EntityByName("minecraft:splash_potion")
+	if !ok {
+		return false
+	}
+
+	p, ok := splash.(projectile)
```

Comment:
I removed the `projectile` interface in the latest commit. Make sure to replace that with an interface that has the `New` method following the signature of SplashPotion.New.